lib/smack/SmackInstGenerator.cpp

```diff
@@ -623,6 +623,16 @@ void SmackInstGenerator::visitCallInst(llvm::CallInst& ci) {
     Value* val = ci.getArgOperand(0);
     emit(Stmt::assign(rep->expr(&ci), rep->expr(val)));
 
+  } else if (SmackOptions::BitPrecise && SmackOptions::FloatEnabled &&
```

Comment:
Let's pull out handing of intrinsics into a separate function, so that it is not mixed with e.g. handing of annotations. I suggest we refactor this by introducing two private functions `processIntrinsicCall` and `processContractCall` (or something like that).